### PR TITLE
Test transform

### LINKED ISSUE #48: add tests for transform module where applicable

### LINKED ISSUE #53: implement @freeze_time decorator
do we need to "freeze" this for testing?

_Originally posted by @bbelderbos in https://github.com/BenGriffith/cryptocurrency/pull/52#discussion_r1351984304_

### PR DESCRIPTION
**Pull request by BenGriffith**

### CODE REVIEW

File: src/crypto/transform.py [hunk 1]
```diff
@@ -112,7 +112,7 @@ def tag_dim_rows(self, crypto_data: dict) -> list:
             rows.append({"tag_key": key, "tag": tag})
         return rows
 
-    def name_tag_bridge_table(self, date_key: date, crypto_data: dict) -> list:
+    def name_tag_bridge_table(self, date_key: str, crypto_data: dict) -> list:
```
Comment (bbelderbos) [inline src/crypto/transform.py:115]: you could further subtype dict and list

File: tests/unit/test_transform_helpers.py [hunk 2]
```diff
@@ -30,8 +33,9 @@ def test_month_dim_rows(transform):
 
 
 def test_date_dim_row(transform, date_dim_rows):
-    rows = transform.date_dim_row()
-    rows == [date_dim_rows]
+    with freeze_time("2023-08-01"):
```
Comment (bbelderbos) [inline tests/unit/test_transform_helpers.py:36]: nice, used to decorator but it has a context manager as well :)

**Review by bbelderbos — COMMENTED** [review]
Looks good

File: tests/unit/test_transform_helpers.py (hunk 2 above)
Comment (BenGriffith) [reply tests/unit/test_transform_helpers.py:36]: I first tried with the decorator but didn't work as expected so I tried with the context manager and worked. My first time working freeze_time so very possible I didn't setup decorator correctly. What made it a little different from documentation scenarios is date is being set within the `date_dim_row` method of the test function ... not explicitly within the test function like `test_date = datetime(...)` - this is what made me try out the context manager after trying the decorator

**Pull request merged**